pkg/services/ngalert/state/manager.go

```diff
@@ -150,18 +149,54 @@ func (st *Manager) Get(orgID int64, alertRuleUID, stateId string) *State {
 	return st.cache.get(orgID, alertRuleUID, stateId)
 }
 
-// ResetStateByRuleUID deletes all entries in the state manager that match the given rule UID.
-func (st *Manager) ResetStateByRuleUID(ctx context.Context, ruleKey ngModels.AlertRuleKey) []*State {
-	logger := st.log.New(ruleKey.LogContext()...)
+// ResetStateByDeletedRuleUID deletes all entries in the state manager that match the given rule UID.
+func (st *Manager) ResetStateByDeletedRuleUID(ctx context.Context, rule *ngModels.AlertRule) []*State {
+	return st.resetStateByRuleUID(ctx, rule, ngModels.StateReasonDeleted)
+}
+
+// ResetStateByPausedRuleUID deletes all entries in the state manager that match the given rule UID.
+func (st *Manager) ResetStateByPausedRuleUID(ctx context.Context, rule *ngModels.AlertRule) []*State {
+	return st.resetStateByRuleUID(ctx, rule, ngModels.StateReasonPaused)
+}
+
+func (st *Manager) resetStateByRuleUID(ctx context.Context, rule *ngModels.AlertRule, reason string) []*State {
+	key := rule.GetKey()
+	logger := st.log.New(key.LogContext()...)
 	logger.Debug("Resetting state of the rule")
-	states := st.cache.removeByRuleUID(ruleKey.OrgID, ruleKey.UID)
+
+	states := st.cache.removeByRuleUID(rule.OrgID, rule.UID)
 	if len(states) > 0 && st.instanceStore != nil {
-		err := st.instanceStore.DeleteAlertInstancesByRule(ctx, ruleKey)
+		err := st.instanceStore.DeleteAlertInstancesByRule(ctx, key)
 		if err != nil {
 			logger.Error("Failed to delete states that belong to a rule from database", "error", err)
 		}
 	}
-	logger.Info("Rules state was reset", "states", len(states))
+
+	if reason == ngModels.StateReasonPaused {
+		transitions := make([]StateTransition, 0, len(states))
+		for _, s := range states {
+			oldState := s.State
+			oldReason := s.StateReason
+			s.State = eval.Normal
```

Comment:
I think you can use `state.SetNormal(reason, s.StartsAt, s.EndsAt)` here instead?